### ISSUE TITLE
Duplicated values in `CMAKE_C_FLAGS` at fresh run

### ISSUE THREAD

**Issue by approxit** [AmigaPorts/AmigaCMakeCrossToolchains#9]
Few days ago I've spend some time with @tehKaiN on debugging my amiga dev setup and its quirks.
By constantly rejecting layers upon layers, we came to point that my code was oddly compiled with duplicated flags passed to compiler by cmaketoolchain.

Setting [`CMAKE_C_FLAGS`](https://github.com/AmigaPorts/AmigaCMakeCrossToolchains/blob/master/m68k-amigaos.cmake#L77) is actually appending value to itself.

For some reason toolchain config runs twice, and `CMAKE_C_FLAGS` have duplicated `FLAGS_COMMON` value, which causes errors for me.

In addition, fresh start of cmake is affected, but next cmake calls don't have duplicated flags.
Ideally I'd like to run cmake once, and receive desired output. Is something with cmaketoolchain, or I'm missing few things?

**Comment by MBeijer** [comment]
@approxit hmm, could you provide a PR with a fix for this?

**Comment by simpsond** [comment]
The toolchain config runs multiple times on a fresh start, for sure and this is normal as CMake does compiler checks.  I believe the problem here is that CMAKE_\<LANG\>_FLAGS is passed through from check to check<sup>1</sup>.

Running a fresh cmake, `cmake -B cmake-m68k -DCMAKE_TOOLCHAIN_FILE=m68k-amigaos.cmake -DTOOLCHAIN_PATH=/opt/amiga-gcc -DCMAKE_VERBOSE_MAKEFILE=1`

and then `make` gives the duplicated flags
```
[ 50%] Building C object CMakeFiles/AmiHello.dir/hello.c.obj
/opt/amiga-gcc/bin/m68k-amigaos-gcc --sysroot=/opt/amiga-gcc   -m68000 -msoft-float -fomit-frame-pointer -mcrt=nix20   -m68000 -msoft-float -fomit-frame-pointer -mcrt=nix20  -std=gnu99 -MD -MT CMakeFiles/AmiHello.dir/hello.c.obj -MF CMakeFiles/AmiHello.dir/hello.c.obj.d -o CMakeFiles/AmiHello.dir/hello.c.obj -c /Volumes/workspace/amiga_dev/hello_world/hello.c
```

if you then run the cmake command again and then make you don't get the duplicates any longer.
```
[ 50%] Building C object CMakeFiles/AmiHello.dir/hello.c.obj
/opt/amiga-gcc/bin/m68k-amigaos-gcc --sysroot=/opt/amiga-gcc   -m68000 -msoft-float -fomit-frame-pointer -mcrt=nix20  -std=gnu99 -MD -MT CMakeFiles/AmiHello.dir/hello.c.obj -MF CMakeFiles/AmiHello.dir/hello.c.obj.d -o CMakeFiles/AmiHello.dir/hello.c.obj -c /Volumes/workspace/amiga_dev/hello_world/hello.c
```

<sup>1.</sup> Scott C. (2018 - 2022) _[Professional CMake: A Practical Guide](https://crascit.com/professional-cmake/)_

**Comment by simpsond** [comment]
I'm wondering in what use-case does the CMAKE_\<LANG\>_FLAGS need to be in the set string, e.g., `set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${FLAGS_COMMON} ${TOOLCHAIN_CFLAGS}")`

Wouldn't `set(CMAKE_C_FLAGS "${FLAGS_COMMON} ${TOOLCHAIN_CFLAGS}")` be fine?

Happy to do a PR if that is all there is to it.

**Comment by tehKaiN** [comment]
I think it's mostly for being nice and not removing default cmake compiler flags, if there are any.

**Comment by simpsond** [comment]
Actually I wonder if we should be using the _INIT version of these variables, see: https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_FLAGS_INIT.html

Again citing Scott C. (2018 - 2022) [Professional CMake: A Practical Guide](https://crascit.com/professional-cmake/)
“Developers can add their own flags into the set of default values for these using variables of the same name but with _INIT appended. These …​_INIT variables are only ever used to set the initial defaults, they have no effect once CMake has been run once and the actual values have been saved in the cache.

**_A common mistake is to set the non-…​INIT variables in a toolchain file (i.e. setting CMAKE_<LANG>_FLAGS rather than CMAKE_<LANG>_FLAGS_INIT). This has the undesirable effect of discarding or hiding any changes the developer might make to these variables in the cache._** The toolchain file may also be re-read as part of later project() or enable_language() calls, thereby discarding any changes to these variables made by the project itself. Setting the …​INIT variables instead ensures that only the initial default values are affected and any subsequent changes to the non-…​_INIT variables via any method are retained.”

I tested, https://github.com/simpsond/AmigaCMakeCrossToolchains/blob/435fad31d21881b25a1bceb538b24b2f4a00e17b/m68k-amigaos.cmake and it seems to work.

**Comment by MBeijer** [comment]
Solved with #16
